### PR TITLE
266 idrac_network.py fails with `AttributeError: 'NoneType' object has no attribute 'iDRAC'`

### LINKED ISSUE #160: idrac_bios : 'NoneType' object has no attribute 'BIOS'
Hi

When I try to use module "idrac_bios" for change paramater on my server, i have this error

`"module_stderr": "172.18.94.165 : 'NoneType' object has no attribute 'BIOS'\n172.18.94.165 : 'NoneType' object has no attribute 'BIOS'\
`

My playbook is

```
- name: "Bios Parameter"
      idrac_bios:
        idrac_ip: "{{ idrac_ip }}"
        idrac_password: "{{ idrac_password }}"
        idrac_user: "{{ idrac_user }}"
        attributes:
          #DramRefreshDelay: "Performance"
          #ProcVirtualization: "Enabled"
          #ProcCores: "All"
          #ProcPwrPerf: "MaxPerf"
          #MemFrequency: "MaxPerf"
          SysProfile: "PerfOptimized"
```

I use Release 2.1.2 and omsdk-1.2.445 with ansible 2.9.7

My server is PowerEdge R640 

- BIOS Version | 2.8.1
- iDRAC Firmware Version | 4.22.00.00

Thanks for your help

### LINKED ISSUE #194: module dellemc.openmanage.idrac_network use share_name as mandatory
Hi, I'm new to use the openmanage modules and met this issue. When I wanted to use module dellemc.openmanage.idrac_network to update some idrac network settings like idrac name and dns domain etc, then I got error:

fatal: [localhost]: FAILED! => {"changed": false, "msg": "missing required arguments: share_name"}

But I don't use share at all. so why share_name is needed as mandatory in module dellemc.openmanage.idrac_network?

### LINKED ISSUE #266: idrac_network.py fails with `AttributeError: 'NoneType' object has no attribute 'iDRAC'`
When attempting to update the idrac IP with input:

```
{
    "ANSIBLE_MODULE_ARGS": {
        "idrac_ip": "192.168.1.63",
        "idrac_user": "root",
        "idrac_password": "mypassword",
        "share_name": "\\\\192.168.1.156\\sambashare",
        "share_user": "grant",
        "share_password": "mypassword",
        "ip_address": "192.168.1.64"
    }
}
```

I receive:

```
  File "/home/grant/programs/dellemc-openmanage-ansible-modules/plugins/modules/idrac_network.py", line 399, in main
    msg = run_idrac_network_config(idrac, module)
  File "/home/grant/programs/dellemc-openmanage-ansible-modules/plugins/modules/idrac_network.py", line 309, in run_idrac_network_config
    idrac.config_mgr.configure_ipv4(
  File "/home/grant/.local/lib/python3.8/site-packages/omdrivers/lifecycle/iDRAC/iDRACConfig.py", line 4230, in configure_ipv4
    self.iDRAC_IPv4.Address_IPv4.set_value(ip_address)
  File "/home/grant/.local/lib/python3.8/site-packages/omdrivers/lifecycle/iDRAC/iDRACConfig.py", line 4176, in iDRAC_IPv4
    return self._sysconfig.iDRAC.IPv4
AttributeError: 'NoneType' object has no attribute 'iDRAC'
```

I've run the error down and it occurs because my instance of `omdrivers.lifecycle.iDRAC.iDRACConfig.iDRACConfig` has no value `_sysconfig` in:

```
@property
def iDRAC_IPv4(self):
    return self._sysconfig.iDRAC.IPv4
```

### PR DESCRIPTION
**Pull request by grantcurell**
README.md:
- Added TOC
- Based on user feedback took @anupamaloke 's answer and provided an explanation for the required shares along with how to easily handle the share
- Changed the header Playbooks to Sample Playbooks so user's would have a better idea of what was there

dellemc_configure_idrac_eventing.py
dellemc_configure_idrac_services.py
dellemc_idrac_lc_attributes.py
dellemc_system_lockdown_mode.py
idrac_bios.py
idrac_firmware.py
idrac_network.py
idrac_syslog.pyidrac_timezone_ntp.py:
- Added error handling such that if a share is used and not valid the module gracefully fails.
- Fixes #266
- Fixes #194
- Fixes #160

All files:
- optimized imports

Signed-off-by: Grant Curell <grant_curell@dell.com>

### CODE REVIEW

**Review by jagadeeshnv — CHANGES_REQUESTED** [review]
@grantcurell 
Apologies for the delay. The content looks fine, but need these changes.
Please change the title to something like "Handling invalid share and unused imports cleanup" or whichever is more appropriate considering all the modules modified in this PR.
As a first set of comments.
1. The README.md changes can be a separated from this PR for now.
2.  Please update the content to the latest
3. We will review the message and get back on it

**Comment by jagadeeshnv** [comment]
@grantcurell Please update the file _**tests/unit/plugins/modules/test_idrac_firmware.py**_, then the checks will pass.

**Comment by grantcurell** [comment]
Apologies - I'll be traveling tomorrow. I'll take a look Monday and make the changes

**Comment by grantcurell** [comment]
> @grantcurell
> Apologies for the delay. The content looks fine, but need these changes.
> Please change the title to something like "Handling invalid share and unused imports cleanup" or whichever is more appropriate considering all the modules modified in this PR.
> As a first set of comments.
> 
> 1. The README.md changes can be a separated from this PR for now.
> 2. Please update the content to the latest
> 3. We will review the message and get back on it
> 4. Please exclude ome and redfish modules from this PR and lets keep it to idrac only

I just want to make sure I understand correctly, the request is to delete the changes to all the files in the commit except idrac_network.py and *remove* the fixes for #194 and #160 and leave both of those tickets open?

**Comment by jagadeeshnv** [comment]
@grantcurell 

- Please retain the changes to all iDRAC modules  - idrac_*
- Delete the README and ome_* modules from this PR.(Preferably move to a new PR)
- Change the message as below for the iDRAC modules
```
module.fail_json(msg="Unable to access the share. Ensure that the share name, "
                             "share mount, and share credentials provided are correct.")
```

**Comment by jagadeeshnv** [comment]
@grantcurell Can you enable "Allow edits from maintainers.” so that we can update  `tests/unit/plugins/modules/test_idrac_firmware.py` for the checks to pass and the merge this PR

**Review by jagadeeshnv — APPROVED** [review]

File: plugins/modules/idrac_firmware.py [hunk 1]
```diff
@@ -411,6 +411,12 @@ def update_firmware_omsdk(idrac, module):
             upd_share = FileOnShare(remote="{0}{1}{2}".format(share_name, os.sep, catalog_file_name),
                                     mount_point=module.params['share_mnt'], isFolder=False,
                                     creds=UserCredentials(share_user, share_pwd))
+
+            if not upd_share.IsValid:
```
Comment (rajeevarakkal) [inline plugins/modules/idrac_firmware.py:415]:
```suggestion
            if not upd_share.IsValid:
```

double entry

**Review by jagadeeshnv — DISMISSED** [review]

**Comment by jagadeeshnv** [comment]
@grantcurell 
`plugins/modules/idrac_firmware.py` had issues with indentation and duplicate if conditions and UT checks for this was failing as UT file was not committed. We tried to exclude idrac_firmware.py from the commit, Since our release activity is in progress today(6/24)

Can you please fix the issues or restore the idrac_firmware module without any changes so that checks will pass.
For your quick reference, PR #288 has the required changes.

**Comment by grantcurell** [comment]
Apologies for not being more responsive. I took on a new project and am working pretty well at max capacity. I'll be out of town tomorrow but I'll take a look when I'm at the airport Monday

**Review by rajeevarakkal — APPROVED** [review]

**Review by felixs88 — APPROVED** [review]

**Comment by rajeevarakkal** [comment]
> Apologies for not being more responsive. I took on a new project and am working pretty well at max capacity. I'll be out of town tomorrow but I'll take a look when I'm at the airport Monday

No problem,   since our release is due, we will merge now and we will push the necessary UT correction immediately

**Pull request merged**